sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/SarifV1JSONImporter.java

```diff
@@ -0,0 +1,187 @@
+package com.daimler.sechub.sereco.importer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+import com.contrastsecurity.sarif.SarifSchema210;
+import com.contrastsecurity.sarif.ThreadFlow;
+import com.contrastsecurity.sarif.CodeFlow;
+import com.contrastsecurity.sarif.Location;
+import com.contrastsecurity.sarif.PhysicalLocation;
+import com.contrastsecurity.sarif.ReportingDescriptorReference;
+import com.contrastsecurity.sarif.Result;
+import com.contrastsecurity.sarif.Result.Level;
+import com.daimler.sechub.commons.model.ScanType;
+import com.daimler.sechub.sereco.metadata.SerecoCodeCallStackElement;
+import com.daimler.sechub.sereco.metadata.SerecoMetaData;
+import com.daimler.sechub.sereco.metadata.SerecoSeverity;
+import com.daimler.sechub.sereco.metadata.SerecoVulnerability;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+/** 
+ * This Importer supports SARIF Version 2.1.0
+ * https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html
+ *  */
+@Component
+public class SarifV1JSONImporter extends AbstractProductResultImporter {
+
+    public SerecoMetaData importResult(String data) throws IOException {
+
+        if (data == null) {
+            data = "";
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        SarifSchema210 sarif;
+
+        try {
+            sarif = mapper.readValue(data, SarifSchema210.class);
+        } catch (Exception e) {
+            throw new IOException("Import cannot parse sarif 2.1.0 json", e);
+        }
+
+        SerecoMetaData metaData = new SerecoMetaData();
+
+        sarif.getRuns().stream().forEach(run -> {
+            
+            List<Result> results = resultsWithGroupedLocations(run.getResults());
+            
+            results.stream().forEach(result -> {
+                SerecoVulnerability vulnerability = new SerecoVulnerability();
+
+                vulnerability.setDescription("");
```

Comment:
Why is the description always empty?

In SARIF the description can be empty. However, you should expect a result to have a [message object](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317459), [short description](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317552) or [full description](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317553).  All of those should have at least a `text` attribute.

The SARIF reports which Brakeman creates use the `message` property. The example results from the SARIF tutorial use the massage object as well: https://github.com/microsoft/sarif-tutorials/blob/main/docs/2-Basics.md#results.

Furthermore, the `text` property can have placeholders (see: [Messages with arguments](https://github.com/microsoft/sarif-tutorials/blob/main/docs/3-Beyond-basics.md#messages-with-arguments) and [Messages with placeholders](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317466)). In addition, the `text` property can contain hyperlinks formatted in markdown (see: [Messages with embedded links](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317467)).

You can ignore the placeholder arguments and links for now. The hyperlinks in markdown can easily be understood by users. I am not aware of any tool making use of the placeholder arguments for now. However, having a description of the vulnerability is important.

Reply:
This is analog to the behavior of `CheckmarxV1XMLImporter.java`. No description is set there too. Because of the comment, I thought this is the desired behavior for some tools.

![image](https://user-images.githubusercontent.com/71753812/109153348-173dd180-776d-11eb-8cb6-cd3b8022fbc4.png)


I think we need to assess all the tools we will be using to generate SARIF reports before we fill the description field in the future. Some tools could be exposing proprietary information which could be confusing or misleading for our SecHub users.

Reply:
@4w31ss3 I was assuming that is the place where we set the description, but apparently not. 

@de-jcup please have a look at this. Maybe, you can explain that.

Reply:
@Jeeppler , @4w31ss3 : `CheckmarxV1XMLImporter` does set description to empty - that is correct (and ugly). The reason was that at time of implementation checkmarx always returned only full HTML. So we were not able to set normal description text (about the problem) without formatting etc.

When SARIF defines description as plain text we should always reuse the description and just set it. If this cannot be ensured - e.g. some tools integrate HTML here, we could (maybe in a future request) do a filtering and only convert into our own description fields at import time when format is accepted (e.g. asciidoc/markdown ... )

**Sum up:** I would think we should set the description. Filtering unwanted content should be done as well (but in a later issue)

Reply:
@4w31ss3 : Please create a new github issue for setting the description field and link it inside the #386  issue. 
@Jeeppler : we must inspect content from different SARIF messages etc. and the implementation to check which messages and how to filter content is not so trivial. So we let this issue as is and do not set the description here. In a leter issue this will be handled

Reply:
@de-jcup: I created #575 for setting of the description field.